tests/sympc/grad/grad_functions_test.py

```diff
@@ -814,3 +815,53 @@ def test_grad_maxpool_2d_backward(get_clients, kernel_size, stride, padding) ->
 
     res = res_mpc.reconstruct()
     assert np.allclose(res, expected_grad, rtol=1e-3)
+
+
+def test_grad_div_forward(get_clients) -> None:
+    # We need Function Secret Sharing (only for 2 parties) for
+    # comparing
+    parties = get_clients(2)
+    x = torch.tensor([2.0, 3.0])
+    y = torch.tensor([6.0, 4.0])
+
+    x_mpc = x.share(parties=parties)
+    y_mpc = y.share(parties=parties)
+
+    ctx = {}
+    res_mpc = GradDiv.forward(ctx, x_mpc, y_mpc)
+
+    res = res_mpc.reconstruct()
+    expected = x / y
+
+    assert np.allclose(res, expected, rtol=1e-3)
+
+
+def test_grad_div_backward(get_clients) -> None:
+    parties = get_clients(2)
+
+    grad = torch.tensor([1, 2])
+
+    session = Session(parties=parties)
+    session.autograd_active = True
+    SessionManager.setup_mpc(session)
+
+    x_secret = torch.tensor([2.0, 3.0])
+    x = MPCTensor(secret=x_secret, session=session, requires_grad=True)
+
+    y_secret = torch.tensor([9.0, 8.0])
+    y = MPCTensor(secret=y_secret, session=session, requires_grad=True)
+
+    grad_mpc = MPCTensor(secret=grad, session=session, requires_grad=True)
+
+    ctx = {"x": x, "y": y}
+
+    grad_x, grad_y = GradDiv.backward(ctx, grad_mpc)
+
+    expected_grad_x = grad * (x_secret / (y_secret * x_secret))
+    expected_grad_y = ((-1) * (x_secret)) / ((y_secret ** 2) * grad)
+
+    res_x = grad_x.reconstruct()
+    res_y = grad_y.reconstruct()
+
+    assert np.allclose(res_x, expected_grad_x, rtol=1e-1)
```

Comment:
Q: Can we have here `1e-2`

Reply:
The tests were failing because there were small differences and the numbers are also small. The output in plaintext was: _tensor([-0.0268, -0.0234])_ and the result after reconstruction:  _tensor([-0.0247, -0.0234])_. Instead of keeping it this way I will change the values being passed (x,y, grad) so that gradients are small enough and we can use 1e-2.